pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -591,6 +590,10 @@ func (e *Controller) checkLeftoverEndpoints() {
 	}
 }
 
+// addEndpointSubset add the endpoints addresses and ports to the EndpointSubset.
+// The addresses are added to the corresponding field, ready or not ready, depending
+// on the pod status and the Service PublishNotReadyAddresses field value.
+// The pod passed must belong to the Endpoint.
```

Comment:
I think the key thing that needs to be highlighted here is that endpoints passed to this function _must_ have already been filtered through `ShouldPodBeInEndpointSlice`.